src/Eloquent/HybridRelations.php

```diff
@@ -328,6 +334,139 @@ public function belongsToMany(
         );
     }
 
+    /**
+     * Define a morph-to-many relationship.
+     *
+     * @param  string $related
+     * @param    string $name
+     * @param  null   $table
+     * @param  null   $foreignPivotKey
+     * @param  null   $relatedPivotKey
+     * @param  null   $parentKey
+     * @param  null   $relatedKey
+     * @param  null   $relation
+     * @param  bool   $inverse
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\MorphToMany
+     */
+    public function morphToMany(
+        $related,
+        $name,
+        $table = null,
+        $foreignPivotKey = null,
+        $relatedPivotKey = null,
+        $parentKey = null,
+        $relatedKey = null,
+        $relation = null,
+        $inverse = false,
+    ) {
+        // If no relationship name was passed, we will pull backtraces to get the
+        // name of the calling function. We will use that function name as the
+        // title of this relation since that is a great convention to apply.
+        if ($relation === null) {
+            $relation = $this->guessBelongsToManyRelation();
+        }
+
+        // Check if it is a relation with an original model.
+        if (! is_subclass_of($related, Model::class)) {
+            return parent::MorphToMany(
+                $related,
+                $name,
+                $table,
+                $foreignPivotKey,
+                $relatedPivotKey,
+                $parentKey,
+                $relatedKey,
+                $relation,
+                $inverse,
+            );
+        }
+
+        $instance = new $related();
+
+        $foreignPivotKey = $foreignPivotKey ?: $name . '_id';
+        $relatedPivotKey = $relatedPivotKey ?:  Str::plural($instance->getForeignKey());
+
+        // Now we're ready to create a new query builder for the related model and
+        // the relationship instances for this relation. This relation will set
+        // appropriate query constraints then entirely manage the hydration.
+        if (! $table) {
+            $words = preg_split('/(_)/u', $name, -1, PREG_SPLIT_DELIM_CAPTURE);
+            $lastWord = array_pop($words);
+            $table = implode('', $words) . Str::plural($lastWord);
+        }
+
+        return new MorphToMany(
+            $instance->newQuery(),
+            $this,
+            $name,
+            $table,
+            $foreignPivotKey,
+            $relatedPivotKey,
+            $parentKey ?: $this->getKeyName(),
+            $relatedKey ?: $instance->getKeyName(),
+            $relation,
+            $inverse,
+        );
```

Comment:
I just found `Str::pluralStudly` which does exactly what we need here. I propose that we inline every default value:

```suggestion
        return new MorphToMany(
            $instance->newQuery(),
            $this,
            $name,
            $table ?? Str::pluralStudly($name);,
            $foreignPivotKey ?? $name . '_id',
            $relatedPivotKey ?? Str::plural($instance->getForeignKey()),
            $parentKey ?? $this->getKeyName(),
            $relatedKey ?? $instance->getKeyName(),
            $relation,
            $inverse,
        );
```

Do you want to open PR on the Laravel repository to propose the same change?
https://github.com/laravel/framework/blob/b79025607e4f1c8d9cec59a998d4e59a0b008b27/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php#L597-L603

Reply:
Yes, I'm going to open a PR. much thanks.

Reply:
My PR on Laravel closed under 6 minutes and that's kinda a new record for me😂
However, I noticed there is some difference in regex expressions used in `morphToMany` and `pluralStudly` methods.
The regex used in `pluralStudly` is `/(.)(?=[A-Z])/u` which splits the value followed by an uppercase letter. but in the `morphToMany` method we have `/(_)/u` regex which splits the value using `_` character.

Reply:
I'm sorry about that. So let's make the MongoDB code as close as possible to Eloquent code. Thanks for investigating.